test/extensions/filters/http/compressor/compressor_filter_speed_test.cc

```diff
@@ -520,6 +575,120 @@ BENCHMARK(compressChunks1024WithZstd)
     ->UseManualTime()
     ->Unit(benchmark::kMillisecond);
 
+static const std::array<CompressionParams, 11> brotli_compression_params = {
```

Comment:
const is not sufficient to avoid this being initialized like a non-pod. You need to use constexpr. I might have been wrong about std::array being an appropriate choice; can you just use a C array like
`static constexpr CompressionParams BrotliCompressionParams[] = ....` 

otherwise you can use CONSTRUCT_ON_FIRST_USE to build complex static const objects.

Reply:
I have tried to use `static constexpr CompressionParams BrotliCompressionParams[] = ....`,
but there is an error says 
```
error: the type 'const Envoy::Extensions::HttpFilters::Compressor::CompressionParams []' of 'constexpr' variable 'Envoy::Extensions::HttpFilters::Compressor::gzip_compression_params' is not literal
```
I guess that it is caused by struct.

Reply:
I think that's cause you have a constructor, which you don't need. This compiles fine:
```
struct Foo {
  int a;
  int b;
  int c;
  int d;
};

static constexpr Foo foo_array[] = {
  {0, 1, 2, 3},
  {4, 5, 6, 7}
};
```

Reply:
OK, it works, thanks!